redfish-core/lib/port.hpp

```diff
@@ -162,12 +165,7 @@ inline void
         return;
     }
 
-    getValidFabricAdapterPath(
```

Comment:
Don't we still need to check if a valid fabric adapter?

Reply:
Ed had a comment on https://gerrit.openbmc.org/c/openbmc/bmcweb/+/61097/9..12/redfish-core/lib/port.hpp#b58.

> This isn't needed. Because we're looking these up by association, there's no need to check if the adapter exists. The assocation wont exist if the adapter doesn't exist.

If we check the fabric adapter, we may not be able to get ports on pcie device / fabric adapter.

Currently we have

1)

```
curl -k -X GET https://${bmc}/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10
{
.......
  "Ports": {
    "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports"
  },
...
}

curl -k -X GET https://${bmc}/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports
{
  "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports",
  "@odata.type": "#PortCollection.PortCollection",
  "Members": [],
  "Members@odata.count": 0,
  "Name": "Port Collection"
}

```

2)
After removing fabric adapter check, we have:

```
curl -k -X GET https://${bmc}/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10
{
.......
  "Ports": {
    "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports"
  },
...
}

 curl -k -X GET https://${bmc}:18080/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports/
{
  "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports",
  "@odata.type": "#PortCollection.PortCollection",
  "Members": [
    {
      "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports/c10_connector0"
    },
    {
      "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports/c10_connector1"
    },
    {
      "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports/c10_connector2"
    },
    {
      "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports/c10_connector3"
    }
  ],
  "Members@odata.count": 4,
  "Name": "Port Collection"
}

```

I'm not sure whether we should hide the connectors like "/redfish/v1/Systems/system/FabricAdapters/pcie_card10/Ports/c10_connector0"

Even if we remove it from Ports collection, those can be still be seen via Cables or connector directly.

```
curl -k -X GET https://${bmc}/redfish/v1/Cables/dp0_cable0
{
...
    "UpstreamPorts": [
      {
        "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports/c10_connector0"
      }
    ],
....
  },

curl -k -X GET https://${bmc}/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports/c10_connector0
{
  "@odata.id": "/redfish/v1/Systems/system/FabricAdapters/motherboard-pcieslot10-pcie_card10/Ports/c10_connector0",
  "@odata.type": "#Port.v1_7_0.Port",
  "Id": "c10_connector0",
  "Location": {
    "PartLocation": {
      "ServiceLabel": "U78DB.ND0.WZS000G-P0-C10-T0"
    }
  },
  "LocationIndicatorActive": false,
  "Name": "c10_connector0"
```